core/src/acc_text/nlg/semantic_graph/utils.clj

```diff
@@ -74,6 +75,30 @@
   (prune-branches semantic-graph (set/difference (into #{} (map :id (rest concepts)))
                                                  (into #{} (map :to relations)))))
 
+(defn remove-condition-nodes [semantic-graph]
+  (let [ids-for-removal (->> semantic-graph
+                             ::sg/concepts
+                             (filter #(contains? #{:condition :if-statement :default-statement} (:type %)))
+                             (map :id)
+                             (set))]
+    (log/debugf "Ids for removal: %s" (pr-str ids-for-removal))
+    (-> semantic-graph
+        (update ::sg/concepts (fn [concepts]
+                                (remove #(contains? ids-for-removal (:id %)) concepts)))
+        (update ::sg/relations (fn [relations]
+                                 (let [start-node (first (filter (fn [{:keys [from to]}] (and (contains? ids-for-removal to)
+                                                                                              (not (contains? ids-for-removal from))))
```

Comment:
will this work for multiple condition nodes?

Reply:
Ok, got case when this does fail. Fixing it